clang/test/Driver/cbstring-simple.c

```diff
@@ -0,0 +1,79 @@
+// Comprehensive test for /cbstring flag functionality with all edge cases
+// RUN: %clang_cl /cbstring /c %s -o %t_cbstring.obj
+// RUN: %clang_cl /c %s -o %t_normal.obj  
+// RUN: llvm-objdump -t %t_cbstring.obj | FileCheck %s --check-prefix=CBSTRING
+// RUN: llvm-objdump -t %t_normal.obj | FileCheck %s --check-prefix=NORMAL
+
+// Test comprehensive string literal placement with different scopes and types
+
+#include <stdio.h>
+#include <wchar.h>
+
+// Global scope strings
+const char *global_str = "global_test";
+static const char *static_global_str = "static_global_test";
+
+// String arrays
+const char *array[] = {"array_element1", "array_element2"};
+
+// Wide strings
+const wchar_t *wide_str = L"wide_string_test";
+
+// Function with various string types
+void test_function() {
+    const char *local_str = "local_test";
+    static const char *static_local_str = "static_local_test";
+    
+    // Concatenated strings
+    const char *concat = "concat_" "string_test";
+    
+    // Use strings to prevent optimization
+    printf("%s %s %s\n", local_str, static_local_str, concat);
+}
+
+int main() {
+    const char *main_str = "main_function_test";
+    static const char *static_main_str = "static_main_test";
+    
+    // Use some strings to prevent optimization
+    printf("%s %s %s\n", global_str, array[0], main_str);
+    test_function();
+    
+    return 0;
+}
+
+// With /cbstring flag, strings should be in .text sections
+// CBSTRING: .text
```

Comment:
If you change the `FileCheck` command line above to have `--check-prefixes=NORMAL,CHECK` and `--check-prefix=CBSTRING,CHECK` then you can do something like:

```
// NORMAL: .rdata
// CBSTRING: .text
// CHECK: global_test
// NORMAL: .rdata
// CBSTRING: .text
// CHECK: array_element1
```

You could even go a bit crazy and leverage the [substitution capture support](https://llvm.org/docs/CommandGuide/FileCheck.html#filecheck-string-substitution-blocks) to be able to use `CHECK` in most places:
```
// NORMAL: [[section|\.rdata]]
// CBSTRING: [[section|\.text]]
// CHECK: global_test
// CHECK: [[section]]
// CHECK: array_element1
```